test/jdk/tools/pack200/PackTestZip64.java

```diff
@@ -85,43 +85,53 @@ static void testPacking() throws IOException {
         cmdsList.add(unpackedFile.getName());
         Utils.runExec(cmdsList);
 
-        // Compare files binary
-        compareTwoFiles(testFile, unpackedFile);
+        if (!testFile.exists()) {
+            throw new IOException("File " + testFile.getName() + " does not exist!");
+        }
+
+        if(!unpackedFile.exists()) {
+            throw new IOException("File " + unpackedFile.getName() + " does not exist!");
+        }
+        // Compare two Jar file
+        compareTwoJarFiles(new JarFile(testFile), new JarFile(unpackedFile));
 
         // Cleaning up generated files
         testFile.delete();
         packedFile.delete();
         unpackedFile.delete();
     }
 
-    static void compareTwoFiles(File src, File dst) throws IOException {
-        if (!src.exists()) {
-            throw new IOException("File " + src.getName() + " does not exist!");
-        }
+    static void compareTwoJarFiles(JarFile srcJar, JarFile dstJar) throws IOException {
+        Enumeration entryList = srcJar.entries();
+        while (entryList.hasMoreElements()) {
+            JarEntry srcEntry = (JarEntry) entryList.nextElement();
+            JarEntry dstEntry = dstJar.getJarEntry(srcEntry.getName());
 
-        if(!dst.exists()) {
-            throw new IOException("File " + dst.getName() + " does not exist!");
-        }
+            if (dstEntry == null) {
+                throw new IOException("Jar Entry "+srcEntry.getName()+" does not exist in "+dstJar.getName());
+            }
 
-        BufferedInputStream srcis, dstis;
-        srcis = new BufferedInputStream(new FileInputStream(src));
-        dstis = new BufferedInputStream(new FileInputStream(dst));
+            BufferedInputStream srcis, dstis;
+            srcis = new BufferedInputStream(srcJar.getInputStream(srcEntry));
+            dstis = new BufferedInputStream(dstJar.getInputStream(dstEntry));
 
-        int s = 0, d, pos = 0;
-        while (s != -1) { // Checking of just one result for EOF is enough
-            s = srcis.read();
-            d = dstis.read();
+            int s = 0, d, pos = 0;
+            while (s != -1) { // Checking of juust one result for EOF is enough
+                s = srcis.read();
+                d = dstis.read();
 
-            if (s != d) {
-                throw new IOException("Files are differ starting at position: "
-                + Integer.toHexString(pos));
-            }
+                if (s != d) {
+                    throw new IOException("Files are differ starting at position: "
+                            + Integer.toHexString(pos));
+                }
 
-            pos++;
+                pos++;
+            }
+            srcis.close();
+            dstis.close();
         }
-
-        srcis.close();
-        dstis.close();
+    srcJar.close();
+    dstJar.close();
```

Comment:
Issues:
* it is _not_ enough to check for EOF once (the 'dst' entry might be longer)
* don't use raw types
* declare each variable separately
* indent consistently

Suggest:
```java
    static void compareTwoJarFiles(JarFile srcJar, JarFile dstJar) throws IOException {
        Enumeration<JarEntry> entryList = srcJar.entries();
        while (entryList.hasMoreElements()) {
            JarEntry srcEntry = entryList.nextElement();
            JarEntry dstEntry = dstJar.getJarEntry(srcEntry.getName());

            if (dstEntry == null) {
                throw new IOException("Jar Entry " + srcEntry.getName() + " does not exist in " + dstJar.getName());
            }

            BufferedInputStream srcis = new BufferedInputStream(srcJar.getInputStream(srcEntry));
            BufferedInputStream dstis = new BufferedInputStream(dstJar.getInputStream(dstEntry));

            for (int pos = 0;; ++pos) {
                int s = srcis.read();
                int d = dstis.read();

                if (s != d) {
                    throw new IOException("Files are differ starting at position: 0x"
                            + Integer.toHexString(pos));
                } else if (s == -1) {
                	break;
                }
            }

            srcis.close();
            dstis.close();
        }

        srcJar.close();
        dstJar.close();
    }
```

Reply:
Thanks a lot @keithc-ca  for the suggestion. In the initial change for the fix, I reused some of the old code from the existing incorrect compareTwoFiles method. But I agree with your suggestions and have incorporated them in https://github.com/ibmruntimes/openj9-openjdk-jdk11/compare/3232726537abaceda9f08afd1521668f3f8704c4..18b42c0619ab09f595149f0dacc6c70521482ac8